src/Microsoft.AspNet.OData.Shared/Query/ODataQueryOptions.cs

```diff
@@ -24,74 +24,23 @@ namespace Microsoft.AspNet.OData.Query
     /// Currently this only supports $filter, $orderby, $top, $skip, and $count.
     /// </summary>
     [ODataQueryParameterBinding]
-    public class ODataQueryOptions
+    public partial class ODataQueryOptions
     {
         private static readonly MethodInfo _limitResultsGenericMethod = typeof(ODataQueryOptions).GetMethod("LimitResults");
 
-        private ETag _etagIfMatch;
-
-        private bool _etagIfMatchChecked;
-
-        private ETag _etagIfNoneMatch;
-
-        private bool _etagIfNoneMatchChecked;
-
         private ODataQueryOptionParser _queryOptionParser;
 
         private AllowedQueryOptions _ignoreQueryOptions = AllowedQueryOptions.None;
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="ODataQueryOptions"/> class based on the incoming request and some metadata information from
-        /// the <see cref="ODataQueryContext"/>.
-        /// </summary>
-        /// <param name="context">The <see cref="ODataQueryContext"/> which contains the <see cref="IEdmModel"/> and some type information.</param>
-        /// <param name="request">The incoming request message.</param>
-        public ODataQueryOptions(ODataQueryContext context, HttpRequestMessage request)
-        {
-            if (context == null)
-            {
-                throw Error.ArgumentNull("context");
-            }
-
-            if (request == null)
-            {
-                throw Error.ArgumentNull("request");
-            }
-
-            // Set the request container into context
-            Contract.Assert(context.RequestContainer == null);
-            context.RequestContainer = request.GetRequestContainer();
-
-            // Remember the context and request
-            Context = context;
-            Request = request;
-
-            // Parse the query from request Uri, including only keys which are OData query parameters or parameter alias
-            RawValues = new ODataRawQueryOptions();
-            IDictionary<string, string> queryParameters = GetODataQueryParameters();
-
-            _queryOptionParser = new ODataQueryOptionParser(
-                context.Model,
-                context.ElementType,
-                context.NavigationSource,
-                queryParameters);
-
-            _queryOptionParser.Resolver = request.GetRequestContainer().GetRequiredService<ODataUriResolver>();
-
-            BuildQueryOptions(queryParameters);
-
-            Validator = ODataQueryValidator.GetODataQueryValidator(context);
-        }
-
-        /// <summary>
         ///  Gets the given <see cref="ODataQueryContext"/>
         /// </summary>
         public ODataQueryContext Context { get; private set; }
 
         /// <summary>
         /// Gets the request message associated with this instance.
         /// </summary>
-        public HttpRequestMessage Request { get; private set; }
+        internal IWebApiRequestMessage InternalRequest { get; private set; }
 
```

Comment:
are we enforcing sort based on public/internal/private? if so, internal comes after public.